controllers/datadogagent/feature/cspm/feature.go

```diff
@@ -0,0 +1,232 @@
+		}
+
+		// TODO add settings to configure f.createSCC and f.createPSP
+
+		reqComp = feature.RequiredComponents{
+			ClusterAgent: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+			Agent: feature.RequiredComponent{
+				IsRequired: apiutils.NewBoolPointer(true),
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.SecurityAgentContainerName,
+				},
+			},
+		}
+	}
+
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *cspmFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+
+	if dda.Spec.Agent.Security != nil && *dda.Spec.Agent.Security.Compliance.Enabled {
+		f.enable = true
+		f.serviceAccountName = v1alpha1.GetClusterAgentServiceAccount(dda)
+
+		if dda.Spec.Agent.Security.Compliance.CheckInterval != nil {
+			f.checkInterval = strconv.FormatInt(dda.Spec.Agent.Security.Compliance.CheckInterval.Duration.Nanoseconds(), 10)
+		}
+
+		if dda.Spec.Agent.Security.Compliance.ConfigDir != nil {
+			f.configMapName = dda.Spec.Agent.Security.Compliance.ConfigDir.ConfigMapName
+			f.configMapConfig = v1alpha1.ConvertConfigDirSpec(dda.Spec.Agent.Security.Compliance.ConfigDir).ConfigMap
+		}
+
+		reqComp = feature.RequiredComponents{
+			ClusterAgent: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+			Agent: feature.RequiredComponent{
+				IsRequired: apiutils.NewBoolPointer(true),
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.SecurityAgentContainerName,
+				},
+			},
+		}
+	}
+
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *cspmFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	if f.createSCC {
+		// Manage SecurityContextConstraints
+		sccName := getSCCName(f.owner)
+		scc, err := managers.PodSecurityManager().GetSecurityContextConstraints(f.owner.GetNamespace(), sccName)
+		if err != nil {
+			return err
+		}
+		scc.AllowHostPID = true
+		managers.PodSecurityManager().UpdateSecurityContextConstraints(scc)
+	}
+
+	if f.createPSP {
+		// Manage PodSecurityPolicy
+		pspName := getPSPName(f.owner)
+		psp, err := managers.PodSecurityManager().GetPodSecurityPolicy(f.owner.GetNamespace(), pspName)
+		if err != nil {
+			return err
+		}
+		psp.Spec.HostPID = true
+		managers.PodSecurityManager().UpdatePodSecurityPolicy(psp)
+	}
+
+	// Manage RBAC
+	rbacName := getRBACResourceName(f.owner)
+
+	return managers.RBACManager().AddClusterPolicyRules("", rbacName, f.serviceAccountName, getRBACPolicyRules())
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *cspmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	cmVol, cmVolMount := volume.GetConfigMapVolumes(
```

Comment:
Will `f.configMapName/Config` always be non-empty values? If not, should this be mounted only if `f.configMapName/Config` aren't empty?